alacritty_terminal/src/tty/unix.rs

```diff
@@ -154,7 +155,10 @@ pub fn new<C>(config: &Config<C>, size: &SizeInfo, window_id: Option<usize>) ->
 
         Shell::new_with_args("/bin/bash", argv)
     } else {
-        Shell::new(pw.shell)
+        match var("SHELL"){
```

Comment:
The `$SHELL` variable is a thing on macOS too, however Terminal.app isn't interested in it.

Should we not support this on macOS? Is there any official resource telling us we shouldn't? Otherwise we could just support this for all unix platforms.

Reply:
I can't find any resources on the matter, and don't have access too macOS for testing as we may need to implement the "ugly hack" from #1856.

Reply:
As you can see in that, we still use `pw.shell` with the macos workaround:

https://github.com/alacritty/alacritty/pull/1856/files#diff-0655ced7de6892bafa1f0000619c718eR228

So we'd just need to define a `shell_program` variable or something similar before this if/else and then use that instead of the `pw.shell`.